code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/process/LoadBalancerServiceUpdatePostListener.java

```diff
@@ -106,18 +107,26 @@ private void updateServicePorts(Service service, ProcessState state) {
                 port = objectManager.create(port);
             }
 
-            allocatorService.allocatePortsForInstanceUpdate(instance, toCreate);
-            
-            for (Port port : toRetain.values()) {
-                createThenActivate(port, new HashMap<String, Object>());
-            }
+            try {
```

Comment:
Nit: it would be better for visibility and understanding to surround just `allocatePortsForInstanceUpdate` call with try/catch. As neither createThenActivate/deactivateThenRemove/releasePortsForInstanceUpdate can throw this exception